### PR TITLE
Migrate from Create React App to Vite

### PR DESCRIPTION
**Pull request by elise-ng**
Ref oursky/likedao#143

- Migrate toolchain from CRA to Vite
- Remove individual polyfills and adopt alternative rollup plugin

Manually walked though the app and found no apparent issues, maybe deploy to dev and see if anyone spot issue?

### CODE REVIEW

File: react-app/tsconfig.json [hunk 1]
```diff
@@ -1,14 +1,15 @@
 {
   "compilerOptions": {
-    "target": "es5",
+    "target": "ESNext",
```
Comment (hochiw) [inline react-app/tsconfig.json:3]: is this field case insensitive

File: react-app/package.json [hunk 2]
```diff
@@ -81,10 +81,10 @@
...
   },
   "scripts": {
-    "start": "react-scripts start",
-    "build": "react-scripts build",
-    "test": "react-scripts test",
-    "eject": "react-scripts eject",
+    "start": "vite",
+    "build": "tsc && vite build",
+    "preview": "vite preview",
+    "test": "jest",
```
Comment (hochiw) [inline react-app/package.json:86]: nice

File: react-app/index.html [hunk 3]
```diff
@@ -16,6 +16,11 @@
     -->
     <link rel="manifest" href="/manifest.json" />
     <title>LikeDAO</title>
+    <script>
+      if (typeof global === "undefined") {
+        var global = window;
+      }
+    </script>
```
Comment (hochiw) [inline react-app/index.html:23]: typeof window?

**Review by hochiw — CHANGES_REQUESTED** [review]

File: react-app/tsconfig.json (hunk 1 above)
Comment (chihimng) [reply react-app/tsconfig.json:3]: i think so but let's align to lower case

File: react-app/index.html (hunk 3 above)
Comment (chihimng) [reply react-app/index.html]: actually let's use `globalThis` instead

**Review by hochiw — APPROVED** [review]

**Pull request merged**